cmake: Move MDK-defines from 'soc' to the MDK build script

SebastianBoe opened the pull request:
The build script ext/hal/nordic/CMakeLists.txt manages the MDK
dependency. The MDK is not aware of Zephyr and uses it's own defines
to indicate what HW we are building for.

This PR moves these MDK-specific defines out of the 'soc' build
scripts and into the MDK build script where they belong.

It also drops support for a define that is unused in-tree and believed to have been
added by accident.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
cmake: Move MDK-defines from 'soc' to the MDK build script

The build script ext/hal/nordic/CMakeLists.txt manages the MDK
dependency. The MDK is not aware of Zephyr and uses it's own defines
to indicate what HW we are building for.

This commit moves these MDK-specific defines out of the 'soc' build
scripts and into the MDK build script where they belong.

This is expected to be a pure refactoring without side-effects.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

anangl commented on soc/arm/nordic_nrf/nrf51/CMakeLists.txt:
```diff
@@ -1,6 +1,7 @@
 zephyr_compile_definitions_ifdef(
   CONFIG_SOC_SERIES_NRF51X
-  NRF51
   NRF51822
```
Now it looks a bit weird. Why `NRF51822` is to stay here? I think it should be also moved. Or better removed. Is it actually used anywhere?

SebastianBoe replied:
There are no known in-tree users.

I can remove it. I believe it was added by mistake and likely never used out-of-tree.

SebastianBoe replied:
Removed.

SebastianBoe committed:
cmake: nrf51: Don't #define NRF51822

The define NRF51822 has no in-tree users and is believed to have been
added by accident. The define NRF51 should be referenced in it's
place.

To avoid users accidentally referencing NRF51822 instead of NRF51, and
to avoid confusing users into believing that NRF51822 has semantics we
remove it.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe commented:
@carlescufi : This is ready to be merged.
